frontend/src/components/viewers/Tensorboard.tsx

```diff
@@ -70,51 +89,167 @@ class TensorboardViewer extends Viewer<TensorboardViewerProps, TensorboardViewer
       <div>
         {this.state.podAddress && (
           <div>
-            <div className={padding(20, 'b')}>Tensorboard is running for this output.</div>
+            <div className={padding(20, 'b')}>{`Tensorboard is running for this output.`}</div>
             <a
               href={'apis/v1beta1/_proxy/' + podAddress}
               target='_blank'
               className={commonCss.unstyled}
             >
-              <Button className={commonCss.buttonAction} disabled={this.state.busy}>
+              <Button
+                className={commonCss.buttonAction}
+                disabled={this.state.busy}
+                style={{ marginBottom: 20, width: 150 }}
+              >
                 Open Tensorboard
               </Button>
             </a>
+
+            <div>
+              <Button
+                className={commonCss.buttonAction}
+                disabled={this.state.busy}
+                id={'delete'}
+                title={`stop tensorboard and delete its instance`}
+                onClick={this._handleDeleteOpen.bind(this)}
+                style={{ marginBottom: 20, width: 150 }}
+              >
+                Delete Tensorboard
+              </Button>
+              <Dialog
+                open={this.state.deleteDialogOpen}
+                onClose={this._handleDeleteClose.bind(this)}
+                aria-labelledby='dialog-title'
+              >
+                <DialogTitle style={{ cursor: 'move' }} id='dialog-title'>
+                  {`Stop Tensorboard ${this.state.tensorflowVersion}?`}
+                </DialogTitle>
+                <DialogContent>
+                  <DialogContentText>
+                    You can stop the current running tensorboard. The tensorboard viewer will also
+                    be deleted from your GKE workloads.
+                  </DialogContentText>
+                </DialogContent>
+                <DialogActions>
+                  <Button
+                    id={'cancel'}
+                    autoFocus={true}
+                    onClick={this._handleDeleteClose.bind(this)}
+                    style={{ width: 50 }}
+                    color='primary'
+                  >
+                    Cancel
+                  </Button>
+                  <BusyButton
+                    id={'confirm'}
+                    className={commonCss.buttonAction}
+                    onClick={this._deleteTensorboard.bind(this)}
```

Comment:
nit: recommend changing this._deleteTensorboard function to an arrow function and remove this bind.

Reply:
Will remove this bind

Reply:
reminder